src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -114,6 +119,16 @@ Version VSVersion
 
         protected override async tasks.Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
+            var environment = new Rothko.Environment();
+            var logPath = Path.Combine(environment.GetLocalGitHubApplicationDataPath(), "extension.log");
+
+            Log.Logger = new LoggerConfiguration()
+                .Enrich.WithThreadId()
+                .WriteTo.File(logPath,
+                    fileSizeLimitBytes: 2L * 1024L * 1024L,
+                    outputTemplate: "{Timestamp:yyyy-MM-dd HH:mm:ss.fff}|{Level}|Thread:{ThreadId}|{SourceContext}|{Message}{NewLine}{Exception}")
+                .CreateLogger();
+
```

Comment:
Yeah, entry points are tricky. There are other areas of the code, particularly the Team Explorer UI, that may be initialized before the package is, they are almost completely separate entities.

Since there are multiple initialization points and all this code needs logging (and currently not everything is using NLog because reasons), what we likely need is a separate class that run this initialization code once and everyone uses it as a gateway to the logging system. If you could move this to such a class and have every caller use that, then we don't have to worry as much about entry points.